client/app/dashboard/setup/preview/page.tsx

```diff
@@ -0,0 +1,189 @@
+"use client";
+import { AccordionManager } from "@/components";
+import { SetupBook, SetupBookshelf, useSetupWizard } from "@/hooks";
+import { getApiClient } from "@/services";
+import { useUser } from "@auth0/nextjs-auth0/client";
+import { Button } from "@jecfe/react-design-system";
+import { useRouter } from "next/navigation";
+import { useEffect, useMemo, useState } from "react";
+import { SetupModal } from "../../SetupModal";
+import { ReviewOption } from "./ReviewOptions";
+
+var setupCustomer = getApiClient()
+  .path("/action/setup-customer")
+  .method("post")
+  .create();
+
+var updateNickname = getApiClient()
+  .path("/customer/update")
+  .method("post")
+  .create();
+
+export default function Preview() {
+  const { isComplete, nickname, books, bookshelves, includeDefaults } =
+    useSetupWizard();
+  const { user } = useUser();
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [nicknameError, setNicknameError] = useState<string | undefined>();
+  const [error, setError] = useState<string | undefined>();
+  const [customerBooks, setCustomerBooks] = useState<SetupBook[]>([]);
+  const [customerNickname, setCustomerNickname] = useState<
+    string | undefined
+  >();
+  const [customerBookshelves, setCustomerBookshelves] =
+    useState<SetupBookshelf>([]);
+  const [customerDefaults, setCustomerDefaults] = useState<boolean>();
+
+  useEffect(() => {
```

Comment:
I hate Next hydration issues. There must be a better way of tackling this -> using this many use effects is a massive code smell